src/HorseProfile/HorseProfile.js

```diff
@@ -1,122 +1,119 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useParams, Link } from "react-router-dom";
 import mockData from "../mockData";
 import "./HorseProfile.scss";
+import { fetchHorse, fetchVet, fetchFarrier, fetchOwner } from '../graphqlQueries.js'
+import { useQuery } from '@apollo/client';
+import ContactDetails from "../ContactDetails/ContactDetails";
+import Form from "../Form/Form";
+import MicroModal from 'micromodal';
 
 function HorseProfile() {
+  const [horse, setHorse] = useState('')
+  const [contacts, setContacts] = useState({
+    farrier: undefined,
+    vet: undefined,
+    owner: undefined,
+  })
   const horseId = Number(useParams().id);
-  let horse;
 
-  mockData.data.horses.forEach((singleHorse) => {
-    if (singleHorse.id === horseId) {
-      horse = singleHorse;
+  useQuery(fetchHorse, {
+    variables: { id: horseId },
+    onCompleted: data => {
+      setHorse(() => data.fetchHorse);
+
     }
   })
 
-  const owner = mockData.data.owners.find((oneOwner) => {
-    return horse.owner_id === oneOwner.id;
-  })
 
-  const vet = mockData.data.vets.find((oneVet) => {
-    return horse.vet_id === oneVet.id;
+  useQuery(fetchFarrier, {
+    variables: { id: horse.farrierId },
+    onCompleted: data => {
+      const newContacts = {...contacts};
+      newContacts.farrier = data.fetchFarrier
+      setContacts(() => newContacts)
+    }
   })
-
-  const farrier = mockData.data.farriers.find((oneFarrier) => {
-    return horse.farrier_id === oneFarrier.id;
+  
+  useQuery(fetchOwner, {
+    variables: { id: horse.ownerId },
+    onCompleted: data => {
+      const newContacts = {...contacts};
+      newContacts.owner = data.fetchOwner
+      setContacts(() => newContacts)
+    }
   })
 
```

Comment:
I think we are missing the fetchVet here? Not hard to add since it's the same format as the others.